components/server/src/projects/projects-service.ts

```diff
@@ -320,6 +320,12 @@ export class ProjectsService {
                 orgId = project.teamId;
                 await db.markDeleted(projectId);
 
+                // delete env vars
+                const envVars = await db.getProjectEnvironmentVariables(projectId);
```

Comment:
nit: Might be worth having an explicit `db.deleteProjectEnvironmentVariables(projectId)` here, just because it performs vastly better. Especially in cases with 10 rows :rolling_eyes:

Reply:
Whoops, meant 10k :sweat_smile: 
I'm suggesting this because I recall there was a very good reason for us to use the default limit of 1000 rows everywhere. Node.js gets very choppy with the amount of bytes to digest, even it does not sound like much.

Reply:
how would you end up with 10k env vars on a project? what 1000 rows limit everywhere do you mean?

Reply:
We synced, and I realized this is the edge case only that we already addressed with the purge migration above.